From 920681d618023f54f4def2eeb0e0f588b2289a92 Mon Sep 17 00:00:00 2001
From: Sebastian Andrzej Siewior <bigeasy@linutronix.de>
Date: Mon, 12 Aug 2019 11:20:44 +0200
Subject: [PATCH 109/251] mm/swap: Enable use pvec lock on RT

On RT we also need to avoid preempt disable/IRQ-off regions so have to enable
the locking while accessing pvecs.

Signed-off-by: Sebastian Andrzej Siewior <bigeasy@linutronix.de>
---
 mm/internal.h | 4 ++++
 mm/swap.c     | 4 ++++
 2 files changed, 8 insertions(+)

diff --git a/mm/internal.h b/mm/internal.h
index 2123fa18cc19..874741f333cf 100644
--- a/mm/internal.h
+++ b/mm/internal.h
@@ -32,7 +32,11 @@
 /* Do not use these with a slab allocator */
 #define GFP_SLAB_BUG_MASK (__GFP_DMA32|__GFP_HIGHMEM|~__GFP_BITS_MASK)
 
+#ifdef CONFIG_PREEMPT_RT
+extern struct static_key_true use_pvec_lock;
+#else
 extern struct static_key_false use_pvec_lock;
+#endif
 
 void page_writeback_init(void);
 
diff --git a/mm/swap.c b/mm/swap.c
index 9afc8767d37b..98accaf51fce 100644
--- a/mm/swap.c
+++ b/mm/swap.c
@@ -44,7 +44,11 @@
 /* How many pages do we try to swap or page in/out together? */
 int page_cluster;
 
+#ifdef CONFIG_PREEMPT_RT
+DEFINE_STATIC_KEY_TRUE(use_pvec_lock);
+#else
 DEFINE_STATIC_KEY_FALSE(use_pvec_lock);
+#endif
 
 struct swap_pagevec {
 	spinlock_t	lock;
-- 
2.19.1

